ddl/ddl_api.go

```diff
@@ -1998,6 +1998,11 @@ func checkTableInfoValidWithStmt(ctx sessionctx.Context, tbInfo *model.TableInfo
 	if err := checkGeneratedColumn(ctx, s.Cols); err != nil {
 		return errors.Trace(err)
 	}
+
+	// Check if table has a primary key if required.
+	if ctx.GetSessionVars().PrimaryKeyRequired && len(tbInfo.GetPkName().String()) == 0 {
+		return infoschema.ErrTableWithoutPrimaryKey
+	}
```

Comment:
I took a look at the internal tables, and many don't have primary keys. Until we can clean it up, we should make this only apply to user-sessions:
```suggestion
	if !ctx.GetSessionVars().InRestrictedSQL && ctx.GetSessionVars().PrimaryKeyRequired && len(tbInfo.GetPkName().String()) == 0 {
		return infoschema.ErrTableWithoutPrimaryKey
	}
```

Reply:
I found this:
```
sql> SET SESSION group_concat_max_len=120;
Query OK, 0 rows affected (0.0003 sec)

sql> SELECT t.TABLE_SCHEMA,COUNT(*),GROUP_CONCAT(t.TABLE_NAME) FROM information_schema.TABLES t LEFT JOIN information_schema.KEY_COLUMN_USAGE kcu ON kcu.TABLE_SCHEMA=t.TABLE_SCHEMA AND kcu.TABLE_NAME=t.TABLE_NAME AND kcu.CONSTRAINT_NAME='PRIMARY' WHERE kcu.CONSTRAINT_NAME IS NULL GROUP BY TABLE_SCHEMA;
+--------------------+----------+--------------------------------------------------------------------------------------------------------------------------+
| TABLE_SCHEMA       | COUNT(*) | GROUP_CONCAT(t.TABLE_NAME)                                                                                               |
+--------------------+----------+--------------------------------------------------------------------------------------------------------------------------+
| PERFORMANCE_SCHEMA |       30 | global_status,session_status,setup_actors,setup_objects,setup_instruments,setup_consumers,events_statements_current,even |
| METRICS_SCHEMA     |      639 | tikv_grpc_avg_req_batch_size,tikv_region_written_bytes,tikv_scheduler_keys_written_avg,tikv_lock_manager_detect_error_to |
| mysql              |       13 | stats_meta,stats_histograms,stats_buckets,gc_delete_range,gc_delete_range_done,stats_feedback,bind_info,stats_top_n,expr |
| INFORMATION_SCHEMA |       76 | OPTIMIZER_TRACE,TIDB_HOT_REGIONS_HISTORY,INSPECTION_SUMMARY,TIFLASH_SEGMENTS,PLUGINS,VIEWS,TIFLASH_REPLICA,CLUSTER_SYSTE |
+--------------------+----------+--------------------------------------------------------------------------------------------------------------------------+
4 rows in set, 1 warning (0.0058 sec)
Warning (code 1260): Some rows were cut by GROUPCONCAT(Column#3)
```
Would we need to add PK's to all of these or just the ones in the `mysql` schema?